src/NuGet.config

```diff
@@ -3,5 +3,6 @@
   <packageSources>
     <clear />
     <add key="dotnet-public" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-public/nuget/v3/index.json" />
+    <add key="dotnet-eng" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json" />
```

Comment:
Just wondering if this is going to be flagged for component governance. We used to get flagged for having two NuGet sources, but is it okay since both are from Microsoft?

Reply:
Can we enable CG on the CI pipeline and confirm that?

Reply:
I'm sure this would get flagged if we merged the change. Long term solution is to create our own public pipeline with all required upstreams. But that come with maintenance costs defining those and managing installed package versions (if we want to do it securely). Alternatively, we can try to get off any packages that we are using from the dotnet-eng feed. Or just keep things the way they are where you have to restore the test project to cache the package before building everything.

Reply:
Did we enable CG and confirm?

It's not a great developer experience to have to restore outside of the IDE in order the build inside the IDE, though it does work. I would also be ok with removing the dependencies on dotnet-eng feed, though in my experiments so far, the conditional testing is pretty deeply ingrained in the tests.

If we can confirm the change will get flagged, I can back it out from this PR.

Reply:
I was wondering how other products like [EF Core](https://github.com/dotnet/efcore/blob/main/NuGet.config) manage multiple feeds in their configuration.. we may want to spend some time understanding the right solution that doesn't involve much dev effort every-time and provides smoother experience to all users.

Reply:
EF Core https://github.com/dotnet/efcore/blob/main/NuGet.config

Reply:
Try this singular feed:

```
<add key="sqlclient" value="https://sqlclientdrivers.pkgs.visualstudio.com/public/_packaging/sqlclient/nuget/v3/index.json" />
```

And then `src/Microsoft.Data.SqlClient/tests/NuGet.config` can probably be deleted.

Reply:
No. CG isn't enabled on branch builds. Only main/production builds.

Reply:
Well ... looks like using that single feed didn't work in the pipelines. Any suggestions?

Reply:
The pipeline just needed a NuGet Authenticate task so that it can install upstream versions of packages to the new feed. I added that and kicked off a new run.

Reply:
@David-Engel it looks like the new builds passed. Feel confident about giving it a approval? 😃